### ISSUE TITLE
Unrecognized selector exception raised when viewing a photo in a support ticket

### ISSUE THREAD

**Issue by hyperspacemark** [zendesk/zendesk_sdk_ios#270]
- SDK version: 1.7.6.1
- Xcode version: 8.2.1
- iOS or Simulator: Reproducable in both
- iOS Version: Reproduced in 10.1.1 and 10.2
- Swift version: 2.3

Steps to Reproduce:

1. Test with a Swift app that **does not** implement the optional `UIApplicationDelegate` property, `var window: UIWindow?`. (I realize this sounds odd but in our application, the window lifecycle is deferred to another object that retains the window and makes it key and visible before launching finishes.)
2. Create and submit a support ticket with a photo attachment.
3. Tap on the attached photo in the submitted ticket.

Expected Result:

The full-screen photo view controller should be presented.

Actual Result:

The application crashes due to an uncaught `NSInvalidArgumentException` as a result of an unrecognized selector, `-[Venmo.AppDelegate window]`, sent to our app delegate instance. The call originates from `+[ZDKUIViewController topViewController]`.

**Comment by eepDev** [comment]
Hi @hyperspacemark, you mentioned that window in `UIApplicationDelegate` is not implemented and that's the exact reason for this crash.

`UIWindow` must be provided to app delegate, there's no workaround for this.
Since you have an object that provides this window, maybe try to wire it somehow to the app delegate.

Thanks,
Elvis

**Comment by hyperspacemark** [comment]
As I mentioned, the `window` requirement on `UIApplicationDelegate` is optional and we'd rather not re-architect when this library could be doing a `-respondsToSelector:` check and falling back to the `keyWindow` on the shared `UIApplication` object.

**Comment by StevenDiviney** [comment]
@hyperspacemark The [documentation for the window](https://developer.apple.com/reference/uikit/uiapplicationdelegate/1623056-window?language=objc) property is a bit fuzzy, saying it's not always required but must return a value. The fact it's marked as optional in the generated swift interface is only really useful in indicating it's wrapping Objective-C code. In any case returning nil is non-standard. I did a quick test and even without a storyboard UIApplicationDelegate will return a window.

I have created an issue for us to track this internally. As we use `[ZDKUIViewController topViewController]` in a few places it will require some regression testing. Unfortunately we are currently in the middle of a release, so I can't give a timeline for when this fix will go out.

Can we get some more information on your setup? Is there a reason your UIApplicationDelegate cannot return a window? We haven't been able to reproduce the issue yet so any information would be appreciated.
